spec/support/factories.rb

```diff
@@ -15,6 +15,18 @@
     author_id '1'
   end
 
+  factory :dynamic_book, class: Book do
+    sequence(:title) do |i|
+      i = i.to_s.rjust(2, "0") if i < 10
+      "Book Title#{i}"
+    end
+    sequence(:description) { |i| "A book about things#{i}" }
+    year '1998'
+    sequence(:user_id) { |i| "#{i}" }
+    sequence(:genre_id) { |i| "#{i}" }
+    sequence(:author_id) { |i| "#{i}" }
```

Comment:
Unless you're intentionally creating a specific `Author`, `User` and `Genre`, you can just do the following, and the associated objects and relationships will be automatically generated:

```
factory :book do
  sequence(:title) { |i| "Book Title#{i}" }
  sequence(:description) { |i| "A book about things#{i}" }
  year '1998'
  user
  genre
  author
end
```